src/Servers/Kestrel/Core/src/Internal/Infrastructure/StringUtilities.cs

```diff
@@ -109,6 +115,257 @@ public static unsafe bool TryGetAsciiString(byte* input, char* output, int count
+                                goto NotEqual;
+                            }
+
+                            offset += sizeof(long);
+                        }
+                        if ((byte*)(offset + sizeof(int)) <= (byte*)count)
+                        {
+                            if (!WidenFourAsciiBytesToUtf16AndComapreToChars(
+                                ref Unsafe.Add(ref str, offset),
+                                Unsafe.ReadUnaligned<uint>(ref Unsafe.Add(ref bytes, offset))))
+                            {
+                                goto NotEqual;
+                            }
+
+                            offset += sizeof(int);
+                        }
+                    }
+                    else
+                    {
+                        // 32-bit: Loop ints by default
+                        while ((byte*)(offset + sizeof(int)) <= (byte*)count)
+                        {
+                            if (!WidenFourAsciiBytesToUtf16AndComapreToChars(
+                                ref Unsafe.Add(ref str, offset),
+                                Unsafe.ReadUnaligned<uint>(ref Unsafe.Add(ref bytes, offset))))
+                            {
+                                goto NotEqual;
+                            }
+
+                            offset += sizeof(int);
+                        }
+                    }
+                    if ((byte*)(offset + sizeof(short)) <= (byte*)count)
+                    {
+                        if (!WidenTwoAsciiBytesToUtf16AndComapreToChars(
+                            ref Unsafe.Add(ref str, offset),
+                            Unsafe.ReadUnaligned<ushort>(ref Unsafe.Add(ref bytes, offset))))
+                        {
+                            goto NotEqual;
+                        }
+
+                        offset += sizeof(short);
+                    }
+                    if ((byte*)offset < (byte*)count)
+                    {
+                        var ch = (char)Unsafe.Add(ref bytes, offset);
+                        if (((ch & 0x80) != 0) || Unsafe.Add(ref str, offset) != ch)
+                        {
+                            goto NotEqual;
+                        }
+                    }
+
+                    // End of input reached, there are no inequalities via widening; so the input bytes are both ascii
+                    // and a match to the string if it was converted via Encoding.ASCII.GetString(...)
+                    return true;
+                }
+
+                // Create a comparision vector for all bits being equal
+                var AllTrue = new Vector<ushort>(ushort.MaxValue);
+                // do/while as entry condition already checked, remaining length must be Vector<byte>.Count or larger.
+                do
+                {
+                    // Read a Vector length from the input as bytes
+                    var vector = Unsafe.ReadUnaligned<Vector<byte>>(ref Unsafe.Add(ref bytes, offset));
```

Comment:
This is still missing the check for "was the input all-ASCII?" The easiest way to do this would be to interpret the byte input as a `Vector<sbyte>` and to ask "is any element less than 0?" See also https://github.com/dotnet/coreclr/blob/56043b6c896417e792e64a69692fed3251ed3adc/src/System.Private.CoreLib/shared/System/Text/ASCIIUtility.cs#L154 for inspiration.

Reply:
Done; had the method just didn't plumb it in O_o